alibi_detect/cd/chisquare.py

```diff
@@ -64,16 +64,20 @@ def __init__(
             input_shape=input_shape,
             data_type=data_type
         )
-        if categories_per_feature is None:  # infer number of possible categories for each feature from reference data
+        # construct categories from the user-specified dict
+        if isinstance(categories_per_feature, dict):
+            vals = list(categories_per_feature.values())
+            if all(isinstance(v, (int, np.int16, np.int32, np.int64)) for v in vals):
```

Comment:
What happens if some values are floats instead of integers, looks like might result in an error later on?? Might be . rather than silently doing inference.

Reply:
It would result in an error which is the desired behaviour I believe. I could make the `NotImplementedError` a bit more informative.

Reply:
I'm not sure this is correct in the case that it is a `dict` of `list`s but the contents of the list are not integers.